mpact_api/mpact/views.py

```diff
@@ -61,19 +63,35 @@ def post(self, request):
                     user_id__id=from_data.get(Constants.ID),
                 )
             except (UserData.DoesNotExist, UserChat.DoesNotExist):
+                logger.debug(
+                    constants.CREATE_UPDATE_MSG,
+                    constants.CREATING,
+                    constants.USER,
+                    from_data,
+                )
                 user_serializer = UserDataSerializer(data=from_data)
                 if user_serializer.is_valid():
                     # Storing user details in both UserData and UserChat Model
                     user_data_rec = UserData.objects.create(**from_data)
                     user_data_rec.save()
+                    logger.debug(constants.DATA_SAVED, constants.USER)
                     return Response(save_userchat(chat_data, user_data_rec))
                 elif user_serializer.errors.get(Constants.ID):
                     # User already exists in user data model but does not exist in userchat model
                     return Response(save_userchat(chat_data, user_instance))
 
             # If user already exists in both the Models then just updating the details(firstname, lastname) in User Data Model
+            logger.debug(
+                constants.CREATE_UPDATE_MSG,
+                constants.UPADTING,
+                constants.USER,
+                from_data,
+            )
             user_serializer = UserDataSerializer(user_instance, data=from_data)
             if user_serializer.is_valid():
                 user_serializer.save()
+                logger.debug(constants.DATA_SAVED, constants.USER)
                 return Response(Constants.OK)
+
+        logger.debug(constants.WEBHOOK_END)
```

Comment:
The `post()` method has three quite distinct if/elif blocks. As the method gets bigger, it makes sense to break the code inside the if/elif blocks out into their own functions. It will make the method more readable.

It will also make it neater when we add exception handling. (I'm guessing the debug logging is temporary, and if `save_chatdata()` fails, for example, instead of just saving a debug log and returning `Response(False)` which I'm sure Telegram doesn't care about, we would raise an exception, and handle it in `post()`, to, maybe, log an error, or save to a model that an admin can follow up on.)

Reply:
Can you please review the changes of PR #3